src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexingService.cs

```diff
@@ -87,14 +113,31 @@ public async Task ProcessContentItemsAsync()
                         return;
                     }
 
+                    var contentManager = scope.ServiceProvider.GetRequiredService<IContentManager>();
+                    var indexHandlers = scope.ServiceProvider.GetServices<IContentItemIndexHandler>();
+
+                    var contentItems = await contentManager.GetAsync(batch.Select(x => x.ContentItemId));
```

Comment:
Here is what I'm considering right now.

```
var contentItems = await contentManager.GetAsync(batch.Select(x => x.ContentItemId), indexSettings.IndexDrafted);
```

So this would return all the "latest" content items. We should maybe consider changing the (bool)IndexDrafted to (bool)IndexLatest.

Reply:
Looks good